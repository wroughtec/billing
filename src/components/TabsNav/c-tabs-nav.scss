.c-tabs-nav {
  display: grid;

  grid-template-columns: repeat(auto-fit, minmax(5rem, 1fr));

  background-color: var(--backgroundColour);
}
