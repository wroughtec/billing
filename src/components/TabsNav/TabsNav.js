// @flow
import React from 'react';
import { tabs } from '../../consts/tabs';
import { Tab } from '../Tab/Tab';
import './c-tabs-nav.scss';

export const TabsNav = () => {
  const tabsNav = tabs.map(tab => {
    const { id, name, url } = tab;

    return <Tab key={id} name={name} url={url} />;
  });

  return <nav className="c-tabs-nav">{tabsNav}</nav>;
};
