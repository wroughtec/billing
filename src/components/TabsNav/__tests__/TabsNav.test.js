import React from 'react';
import { shallow } from 'enzyme';
import { TabsNav } from '../TabsNav';

describe('TabsNav', () => {
  test('Renders', () => {
    const component = shallow(<TabsNav />);

    expect(component).toMatchSnapshot();
  });
});
