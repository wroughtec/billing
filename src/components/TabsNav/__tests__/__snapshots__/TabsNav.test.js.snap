// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsNav Renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabsNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Tab
          name="Bills"
          url="/"
        />,
        <Tab
          name="Transactions"
          url="transactions"
        />,
      ],
      "className": "c-tabs-nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "name": "Bills",
          "url": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "name": "Transactions",
          "url": "transactions",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Tab
            name="Bills"
            url="/"
          />,
          <Tab
            name="Transactions"
            url="transactions"
          />,
        ],
        "className": "c-tabs-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "name": "Bills",
            "url": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "name": "Transactions",
            "url": "transactions",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
